chapters/05-comparison.Rmd

```diff
@@ -27,170 +27,174 @@ distributions_to_evaluate = [
 
-### Accuracy
 
-Plotted below are the comparisons for sample size of $10^4$.
-
-```{python compareZfitKDEpyEstimationsGPU, echo=FALSE, eval=TRUE, fig.cap="Comparison between the newly proposed algorithms 'Binned', 'FFT', 'ISJ' and the FFT based implementation in KDEpy with $n=10^4$ sample points (run on GPU)", fig.align='center', out.width='100%'}
-unrestricted_estimations = pd.read_pickle('./benchmark/gpu_linux/unrestricted_estimations.pkl')
-unrestricted_runtimes = pd.read_pickle('./benchmark/gpu_linux/unrestricted_runtimes.pkl')
-
-figure, axes = runner.plot_estimations(unrestricted_estimations, distributions_to_evaluate, 1e4, ['ZfitBinned', 'ZfitFFT', 'ZfitISJ', 'KDEpyFFT'])
+```{python compareZfitKDEpyRuntimePDF, echo=FALSE, eval=TRUE, fig.cap="Runtime difference of the evaluation phase between the newly proposed algorithms 'Binned', 'FFT', 'ISJ', 'Hofmeyr' and the FFT based implementation in KDEpy", fig.align='center', out.width='100%'}
+figure, axes = runner.plot_runtimes(unrestricted_runtimes, distributions_to_evaluate, ['ZfitBinned', 'ZfitFFT', 'ZfitISJ', 'ZfitHofmeyrK1withCpp', 'KDEpyFFT'], 'pdf')
 figure
 ```
 
-
-```{python compareZfitKDEpyErrorsGPU, echo=FALSE, eval=TRUE, fig.cap="Integrated square errors (ISE) for the newly proposed algorithms 'Binned', 'FFT', 'ISJ' and the FFT based implementation in KDEpy with $n=10^4$ sample points (run on GPU)", fig.align='center', out.width='100%'}
-figure, axes = runner.plot_integrated_square_errors(unrestricted_estimations, distributions_to_evaluate, ['ZfitBinned', 'ZfitFFT', 'ZfitISJ', 'KDEpyFFT'])
+```{python compareZfitRuntimePDFwithoutKDEpy, echo=FALSE, eval=TRUE, fig.cap="Runtime difference of the evaluation phase between the newly proposed algorithms 'Binned', 'FFT', 'ISJ', 'Hofmeyr' only", fig.align='center', out.width='100%'}
+figure, axes = runner.plot_runtimes(unrestricted_runtimes, distributions_to_evaluate, ['ZfitBinned', 'ZfitFFT', 'ZfitISJ', 'ZfitHofmeyrK1withCpp'], 'pdf')
 figure
 ```
 
-Looking at the integrated square errors, we see the same behavior as for the comparison on CPU.
 
-```{python compareZfitKDEpyErrorsGPU8, echo=FALSE, eval=TRUE, fig.cap="Integrated square errors (ISE) for the newly proposed algorithms 'Binned', 'FFT', 'ISJ' and the FFT based implementation in KDEpy", fig.align='center', out.width='100%'}
-figure, axes = runner.plot_integrated_square_errors(unrestricted_estimations, distributions_to_evaluate, ['ZfitBinned', 'ZfitFFT', 'ZfitISJ', 'KDEpyFFT'])
-figure
-```
+Looking only at the evaluation runtimes of the newly proposed methods (figure \@ref(fig:compareZfitRuntimePDFwithoutKDEpy)), we can see that the performance differences are minimimal and even the binned method shows good performance, even if its evaluation runtime increases faster with larger datasets compared to the other implementations.
```

Comment:
I would just remove this, the runtime difference is not really relevant. Maybe the factor, but we already can read all the things off in the plots above. So if, then a statement on the difference/factor, but I would not put a whole 6 plots there for that. just remove